File: app/src/main/java/io/mgba/UI/Activities/SettingsActivity.java
package io.mgba.UI.Activities;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import java.util.HashMap;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.mgba.Adapters.SettingsAdapter;
import io.mgba.Presenter.SettingsPresenter;
import io.mgba.Presenter.Interfaces.ISettingsPresenter;
import io.mgba.R;
import io.mgba.UI.Activities.Interfaces.ISettingsView;
import io.mgba.Utils.IResourcesManager;

public class SettingsActivity extends AppCompatActivity implements ISettingsView {

    @BindView(R.id.recyclerview)
    RecyclerView recyclerView;

    private ISettingsPresenter controller;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        ButterKnife.bind(this);
        controller = new SettingsPresenter(this, (IResourcesManager) getApplication());

        setupToolbar();
        setupRecyclerView();
    }

    private void setupRecyclerView() {
        recyclerView.setHasFixedSize(true);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setAdapter(new SettingsAdapter(controller.getSettings(),
                                                              getApplicationContext(),
                                                              controller.getOnClick(),
                                                              recyclerView));
    }

    private void setupToolbar() {
        if (getSupportActionBar() != null){
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setDisplayShowHomeEnabled(true);
        }
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }

    @Override
    public void startActivity(Class<? extends AppCompatActivity> activity, HashMap<String, String> extras) {
        Intent it = new Intent(this, activity);

        for(String key : extras.keySet())
            it.putExtra(key, extras.get(key));

        startActivity(it);
    }
}

File: app/src/main/java/io/mgba/ui/activities/SettingsActivity.kt
package io.mgba.ui.activities

import android.content.Intent
import android.os.Bundle

import java.util.HashMap

import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import io.mgba.adapters.SettingsAdapter
import io.mgba.presenter.SettingsPresenter
import io.mgba.presenter.interfaces.ISettingsPresenter
import io.mgba.R
import io.mgba.ui.activities.interfaces.ISettingsView
import io.mgba.utilities.IResourcesManager
import kotlinx.android.synthetic.main.activity_settings.*

class SettingsActivity : AppCompatActivity(), ISettingsView {
    private var controller: ISettingsPresenter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)
        controller = SettingsPresenter(this, application as IResourcesManager)

        setupToolbar()
        setupRecyclerView()
    }

    private fun setupRecyclerView() {
        settins_list.setHasFixedSize(true)
        settins_list.layoutManager = LinearLayoutManager(this)
        settins_list.adapter = SettingsAdapter(controller!!.getSettings(),
                applicationContext,
                {controller!!.onClick(it)},
                settins_list)
    }

    private fun setupToolbar() {
        if (supportActionBar != null) {
            supportActionBar!!.setDisplayHomeAsUpEnabled(true)
            supportActionBar!!.setDisplayShowHomeEnabled(true)
        }
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }

    override fun startActivity(activity: Class<out AppCompatActivity>, extras: HashMap<String, String>) {
        val it = Intent(this, activity)

        for (key in extras.keys)
            it.putExtra(key, extras[key])

        startActivity(it)
    }
}
